atmo/jobs/models.py

```diff
@@ -226,3 +213,53 @@ def delete(self, *args, **kwargs):
 
     def get_results(self):
         return self.provisioner.results(self.identifier, self.is_public)
+
+
+@python_2_unicode_compatible
+class SparkJobRun(EditedAtModel):
+
+    spark_job = models.ForeignKey(
+        SparkJob,
+        on_delete=models.CASCADE,
```

Comment:
I'm curious why we might need this? Would keeping a history be useful for calculations?

One idea would be to have a `deleted` flag on the `SparkJob` and update our queries to `SELECT ... WHERE deleted=false` and not actually delete the record but stop showing it to the user if it were deleted. That way we have all the relevant data in the system to perform calculations.